### ISSUE TITLE
StyleGuide - Detach components and directives which are not visible from the view

### ISSUE THREAD

**Issue by johnpapa** [angular/angular.io#1149]
*To be discussed ... *

I copied several of these from early discussions and will list them as issues for future discussion after ngConf

cc // @mgechev @wardbell 
- Detach components and directives which are not visible from the view in order to prevent the change detection running over them.
  
  ``` ts
  /* RECOMMENDED */
  @Directive({
    selector: '[pane]'
  })
  class Pane {
    @Input() title;
    view: EmbeddedViewRef;
    constructor(private _changeDetectorRef: ChangeDetectorRef) {}
    // Attaches the change detector
    show() {
      this._changeDetectorRef.reattach();
    }
    // Detaches the change detector
    hide() {
      this._changeDetectorRef.detach();
    }
  }
  ```

**Comment by zoechi** [comment]
Nice one. 
I saw it mentioned several times that `OnPush` should be favored over the default anyway. If this is the case this should be promoted more in the doc examples as well. 
Would it make sense to detach when `OnPush` is set?

**Comment by wardbell** [comment]
Bad idea. Use other techniques. If temporary, turn change detection off. If you wanted to do what is recommended here, `*ngIf` would do it. I don't think performance optimizations are a matter of style.
